lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -1030,3 +1041,135 @@ unsafeIntCast
 unsafeIntCast x = fromMaybe err $ intCastMaybe x
   where
     err = error $ "unsafeIntCast failed for " <> show x
+
+--------------------------------------------------------------------------------
+-- updateTx
+--------------------------------------------------------------------------------
+
+-- | Describes modifications that can be made to a `Tx` using `updateTx`.
+data TxUpdate = TxUpdate
+    { extraInputs :: [(W.TxIn, W.TxOut)]
+    , extraCollateral :: [W.TxIn]
+       -- ^ Only used in the Alonzo era and later. Will be silently ignored in
+       -- previous eras.
+    , extraOutputs :: [W.TxOut]
+    , extraInputScripts :: [CA.Script CA.KeyHash]
+    , feeUpdate :: TxFeeUpdate
+        -- ^ Set a new fee or use the old one.
+    }
+
+-- | For testing that
+-- @
+--   forall tx. updateTx noTxUpdate tx
+--      == Right tx or Left
+-- @
+noTxUpdate :: TxUpdate
+noTxUpdate = TxUpdate [] [] [] [] UseOldTxFee
+
+-- | Method to use when updating the fee of a transaction.
+data TxFeeUpdate
+    = UseOldTxFee
+        -- ^ Instead of updating the fee, just use the old fee of the
+        -- Tx (no-op for fee update).
+    | UseNewTxFee W.Coin
+        -- ^ Specify a new fee to use instead.
+    deriving (Eq, Show)
+
+newtype ErrUpdateSealedTx
+    = ErrExistingKeyWitnesses Int
+    -- ^ The `SealedTx` couldn't not be updated because the *n* existing
+    -- key-witnesses would have been rendered invalid.
```

Comment:
I find it a bit difficult to understand what this comment means! 😄 

Does the suggested text (below) convey the intended meaning?

```suggestion
    -- ^ The `SealedTx` could not be updated because the *n* existing
    -- key-witnesses would be rendered invalid.
```